lib/osf-components/addon/components/contributor-list/component.ts

```diff
@@ -71,6 +81,46 @@ export default class ContributorList extends Component {
         blocker.done();
     });
 
+    @task({ withTestWaiter: true, enqueue: true })
+    removeMeTask = task(function *(this: ContributorList) {
+        if (!this.node || this.node.isAnonymous || !this.currentUser.currentUserId) {
+            return;
+        }
+
+        const userID = this.currentUser.currentUserId;
+        let contributor = this.displayedContributors
+            .find(contrib => contrib.users.get('id') === this.currentUser.currentUserId);
+
+        if (!contributor) {
+            contributor = yield this.store.findRecord('contributor', `${this.node.id}-${userID}`);
+            this.setProperties({
+                displayedContributors: [...this.displayedContributors, contributor],
+                totalContributors: this.totalContributors + 1,
+            });
+        }
+
+        try {
+            yield contributor!.destroyRecord();
+        } catch (e) {
+            const errorMessage = this.intl.t('contributor_list.remove_contributor.error');
+            captureException(e, { errorMessage });
+            this.toast.error(getApiErrorMessage(e), errorMessage);
+        }
+
+        this.toast.success(this.intl.t('contributor_list.remove_contributor.success'));
+        this.router.transitionTo('home');
+    });
+
+    @action
+    removeMe() {
+        this.removeMeTask.perform();
+    }
+
+    @computed('allowRemoveMe', 'currentUser.currentUserId', 'totalContributors')
+    get shouldShowRemoveMeButton() {
+        return this.allowRemoveMe && this.currentUser.currentUserId && (this.totalContributors || 0) > 1;
```

Comment:
You are initializing totalContributors to 0. Do you need the `|| 0` still?

Reply:
Also, I'm not suggesting to change this, but the downside of not showing the button in the case of only 1 contributor is that the user won't know why the button isn't there. If you had the button and just returned the proper API response when they tried to remove themselves, then they would have more information.